src/org/omegat/Main.java

```diff
@@ -125,126 +75,39 @@ public final class Main {
-
-        if (PARAMS.containsKey(CLIParameters.QUIET)) {
-            RuntimePreferences.setQuietMode(true);
-        }
-
-        if (PARAMS.containsKey(CLIParameters.DISABLE_PROJECT_LOCKING)) {
-            RuntimePreferences.setProjectLockingEnabled(false);
-        }
-
-        if (PARAMS.containsKey(CLIParameters.DISABLE_LOCATION_SAVE)) {
-            RuntimePreferences.setLocationSaveEnabled(false);
-        }
-
-        // initialize logging backend and loading configuration.
-        ILogger logger = Log.getLogger(Main.class);
-
-        logger.atInfo().setMessage("\n{0}\n{1} (started on {2} {3}) Locale {4}")
-                .addArgument(StringUtils.repeat('=', 120)).addArgument(OStrings.getNameAndVersion())
-                .addArgument(DateTimeFormatter.ofLocalizedDate(FormatStyle.MEDIUM)
-                        .withLocale(Locale.getDefault()).format(ZonedDateTime.now()))
-                .addArgument(ZoneId.systemDefault().getDisplayName(TextStyle.SHORT, Locale.getDefault()))
-                .addArgument(Locale.getDefault().toLanguageTag()).log();
-        logger.atInfo().logRB("LOG_STARTUP_INFO", System.getProperty("java.vendor"),
-                System.getProperty("java.version"), System.getProperty("java.home"));
-
         System.setProperty("http.agent", OStrings.getDisplayNameAndVersion());
 
         // Do migration and load various settings. The order is important!
         Preferences.init();
         // broker should be loaded before module loading
         JLanguageTool.setClassBrokerBroker(new LanguageClassBroker());
         JLanguageTool.setDataBroker(new LanguageDataBroker());
-        PluginUtils.loadPlugins(PARAMS);
+        PluginUtils.loadPlugins(null);
         FilterMaster.setFilterClasses(PluginUtils.getFilterClasses());
         Preferences.initFilters();
         Preferences.initSegmentation();
 
-        int result;
-        try {
-            switch (runMode) {
-            case GUI:
-                result = runGUI();
-                // GUI has own shutdown code
-                break;
-            case CONSOLE_TRANSLATE:
-                result = runConsoleTranslate();
-                PluginUtils.unloadPlugins();
-                break;
-            case CONSOLE_CREATEPSEUDOTRANSLATETMX:
-                result = runCreatePseudoTranslateTMX();
-                PluginUtils.unloadPlugins();
-                break;
-            case CONSOLE_ALIGN:
-                result = runConsoleAlign();
-                PluginUtils.unloadPlugins();
-                break;
-            case CONSOLE_STATS:
-                result = runConsoleStats();
-                PluginUtils.unloadPlugins();
-                break;
-            default:
-                result = 1;
-            }
-        } catch (Throwable ex) {
-            Log.log(ex);
-            showError(ex);
-            result = 1;
-        }
-        if (result != 0) {
-            System.exit(result);
-        }
+        commandLine.execute(args);
```

Comment:
It is better to initialize application core after parsing command line arguments.
Current change proposal initialize before parsing argument.